core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -727,6 +727,357 @@ struct MinMaxLoc {
   KOKKOS_INLINE_FUNCTION
   bool references_scalar() const { return references_scalar_v; }
 };
+
+// --------------------------------------------------
+// reducers for std algorithms
+// --------------------------------------------------
+template <class Scalar, class Index, class Space, class ComparatorType>
+struct StdMaxLoc {
+ private:
+  using scalar_type = typename std::remove_cv<Scalar>::type;
+  using index_type  = typename std::remove_cv<Index>::type;
+
+ public:
+  // Required
+  using reducer    = StdMaxLoc<Scalar, Index, Space, ComparatorType>;
+  using value_type = ::Kokkos::ValLocScalar<scalar_type, index_type>;
+
+  using result_view_type = ::Kokkos::View<value_type, Space>;
+
+ private:
+  result_view_type value;
+  bool references_scalar_v;
+  ComparatorType m_comp;
+
+ public:
+  KOKKOS_INLINE_FUNCTION
+  StdMaxLoc(value_type& value_, ComparatorType comp_)
+      : value(&value_), references_scalar_v(true), m_comp(comp_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  StdMaxLoc(const result_view_type& value_, ComparatorType comp_)
+      : value(value_), references_scalar_v(false), m_comp(comp_) {}
+
+  // Required
+  KOKKOS_INLINE_FUNCTION
+  void join(value_type& dest, const value_type& src) const {
+    // complying to std, we can only rely on comp(a,b) returning
+    // true if a is *less* than b
+    if (m_comp(dest.val, src.val)) {
+      dest = src;
+    }
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void join(volatile value_type& dest, const volatile value_type& src) const {
+    if (m_comp(dest.val, src.val)) {
+      dest = src;
+    }
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void init(value_type& val) const {
+    val.val = reduction_identity<scalar_type>::max();
```

Comment:
Is that the right way to initialize regardless of `m_comp`?

Reply:
Thanks for checking! Some of the things you point out are because the work is still in progress :) 
I will reply in a bit!

Reply:
I think this is right for these custom reducers, it is also consistent with those already implemented.